src/praisonai-deploy/tests/test_compose_helm_starters.py

```
@@ -0,0 +1,385 @@
+    env_file = tmp_path / ".env"
+    env_file.write_text(
+        "POSTGRES_PASSWORD= # placeholder\nPOSTGRES_PASSWORD=real-secret-value\n",
+        encoding="utf-8",
+    )
+    _ensure_postgres_password(env_file)
+    lines = [
+        line for line in env_file.read_text(encoding="utf-8").splitlines()
+        if line.strip().startswith("POSTGRES_PASSWORD=")
+    ]
+    # Effective (last) value is non-empty → left untouched.
+    assert any(_env_password_value(l.split("=", 1)[1]) == "real-secret-value" for l in lines)
```

Ruff (0.16.0)

[error] Ambiguous variable name: `l`

(E741)